### DISCUSSION TITLE
Disable `To use retry middleware with Faraday v2.0+, install faraday-retry gem` message

### DISCUSSION THREAD

**Discussion by ShockwaveNN** [octokit/octokit.rb#1486] [Q&A] [answered]
Quite frankly I don't fully understand what this message means

All I know what my app works fine without `faraday-retry` gem

But also show a warning when applications start (which bugs me a little)

I didn't find any info in readme which profits will be if I use `faraday-retry` and if there is none - any way to disable this message

**Comment by skryukov** [comment]
Hey, @ShockwaveNN

`faraday-retry` is a middleware that automatically retries timeouted requests, so it adds some resilience to the app. You can read more about the gem in the [readme](https://github.com/lostisland/faraday-retry).

Yet, if it suits you, you can turn off octokit warnings by setting the ENV `OCTOKIT_SILENT=true`.

**Reply by ShockwaveNN** [reply]
Thanks

Is there any way to disable only this `To use rerty...` warning without disabling any other warning?

**Reply by skryukov** [reply]
nope

**Reply by ShockwaveNN** [reply]
So more robust soltuion will be adding `faraday-retry` as additional dependency?

is there some discussion why it's not octokit forced dependency by default?

**Comment by timrogers** [comment, marked as answer]
If you add `faraday-retry` as a dependency, you’ll get automated retries for failures and the warning will go away.

It isn’t a default dependency for Octokit because we allow users to use an earlier version of Faraday where the retry middleware is included automatically.

It definitely would be possible to introduce a change here to make retries optional, but I had a quick look and it isn’t a trivial code change.

From: Pavel Lobashov ***@***.***>
Date: Tuesday, 20 September 2022 at 07:58
To: octokit/octokit.rb ***@***.***>
Cc: Subscribed ***@***.***>
Subject: Re: [octokit/octokit.rb] Disable `To use retry middleware with Faraday v2.0+, install faraday-retry gem` message (Discussion #1486)

So more robust soltuion will be adding faraday-retry as additional dependency?

is there some discussion why it's not octokit forced dependency by default?

**Reply by ShockwaveNN** [reply]
Thank you for detailed explanation, now I understand why it's not enabled by default and I think I should enable it in my app
